pkg/lexer/lexer.go

```diff
@@ -122,7 +122,7 @@ func lexText(l *lexer) stateFn {
 	case r == '-':
 		l.emit(token.MINUS)
 	case r == '!':
-		l.emit(l.doubleCharOperator('=', token.NOT_LEQ, token.BANG))
+		l.emit(l.doubleCharOperator('=', token.BANG, token.NOT_LEQ))
```

Comment:
Nice catch!

Reply:
Actually, running the tests caught the error :)

Reply:
Haha, I didn't run tests at the end - that's what happens when I'm careless haha 😅